### PR TITLE
Bumps version of main to 2.0.0.0

### LINKED ISSUE #875: Ensure plugins work with Dashboards Node v14.18.2
Dashboards has upgraded Node to v14.18.2 and this is for plugin teams to verify that their plugins work with the new version upgrade. The Dashboards branch running the upgraded node is [feature/node14](https://github.com/opensearch-project/OpenSearch-Dashboards/tree/feature/node14).

A list of things to check is:
* Your plugin's node version is bumped to 14.18.2
* Your plugin's `@types/node` package is bumped to `^14.17.32` if it's used
* Plugin builds and runs on Dashboards
* All of your plugin's tests pass
* Your GitHub workflows are not broken by the node version change and updated if they do
* Any sanity/smoke testing is successful

### PR DESCRIPTION
**Pull request by DarshitChanpura**
### Description
Bumps version to 2.0.0.0

### Issues resolved
* Resolves #875 

### Category
Maintenance

### Check List
- [x] Commits are signed per the DCO using --signoff

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Comment by DarshitChanpura** [comment]
This build requires `2.0` artifact for opensearch-security and hence can only be merged after backend is updated.
See: #1698

**Review by peternied — DISMISSED** [review]

**Comment by DarshitChanpura** [comment]
The test-build seems to be failing because it needs node version upgrade to node@14.18.2, and the PR: #924 is failing because it needs version to be upgraded to 2.0.0.0 which inherently creates a deadlock. How should we proceed @opensearch-project/security ?

**Comment by peternied** [comment]
@DarshitChanpura Can we do both changes in the same PR?

**Comment by DarshitChanpura** [comment]
> This build requires `2.0.0.0` artifact for opensearch-security and hence can only be merged after backend is updated. See: [#1698](https://github.com/opensearch-project/security/pull/1698)

On top of node 14 upgrade, this still stands true. This build requires 2.0.0.0 artifact built for opensearch-security

> @DarshitChanpura Can we do both changes in the same PR?

Okay I will bring node14 upgrade changes in this PR.

File: .github/workflows/unit-test.yml [hunk 1]
```diff
@@ -1,14 +1,5 @@
 name: Unit Tests
 
-on:
```
Comment (peternied) [inline .github/workflows/unit-test.yml:3]: Please update this section to follow suite with the integration tests as well

**Comment by DarshitChanpura** [comment]
I'm trying to debug Unit tests which fail due to mismatch in Jest snapshot matching: See [here](https://github.com/opensearch-project/security-dashboards-plugin/runs/5741154954?check_suite_focus=true#step:12:641)

Meanwhile if you know anything about this error please let me know.

**Comment by DarshitChanpura** [comment]
I'm unable to reproduce it on local. All the unit tests pass when I run them in the same setup. 

Steps that I followed to run the unit tests:
1. Fork core dashboards (`main` branch)
2. Install security plugin (`version-bump-2.0.0.0` branch)
3. Run the tests with this command: `node ./test/run_jest_tests.js --config ./test/jest.config.ui.js`

**Comment by peternied** [comment]
@DarshitChanpura Looks like there are some unit tests failures, Seperately, I'm looking into the integration test failures, I know we will need a code change for at least one of them

**Comment by DarshitChanpura** [comment]
> @DarshitChanpura Looks like there are some unit tests failures, Seperately, I'm looking into the integration test failures, I know we will need a code change for at least one of them

I have fixed unit tests and will push the change soon

**Comment by DarshitChanpura** [comment]
We can re-run the integration tests job once security plugin is added to opensearch-build manifest for 2.0.0

File: .github/workflows/integration-test.yml [hunk 2]
```diff
@@ -15,12 +9,13 @@ jobs:
     steps:
       - name: Download OpenSearch Core
         run: |
-          wget https://ci.opensearch.org/ci/dbc/distribution-build-opensearch/1.3.0/488/linux/x64/builds/opensearch/dist/opensearch-min-1.3.0-linux-x64.tar.gz
+          wget https://ci.opensearch.org/ci/dbc/distribution-build-opensearch/2.0.0/latest/linux/x64/builds/opensearch/dist/opensearch-min-2.0.0-linux-x64.tar.gz
```
Comment (peternied) [inline .github/workflows/integration-test.yml:12]: This build number needs to be updated to 2.0.0-alpha1
```suggestion
          wget https://ci.opensearch.org/ci/dbc/distribution-build-opensearch/2.0.0-alpha1/latest/linux/x64/builds/opensearch/dist/opensearch-min-2.0.0-alpha1-linux-x64.tar.gz
```

**Comment by peternied** [comment]
@DarshitChanpura Note; this can be is still waiting on https://github.com/opensearch-project/opensearch-build/pull/1924 to be merged and a new build has been produced

File: common/index.ts [hunk 3]
```diff
@@ -44,5 +44,6 @@ export enum AuthType {
  */
 export function isValidResourceName(resourceName: string): boolean {
   // see: https://javascript.info/regexp-unicode
-  return !/[\p{C}%]/u.test(resourceName) && resourceName.length > 0;
+  const exp = new RegExp('[p{C}%]', 'u');
```
Comment (peternied) [inline common/index.ts:47]: I think this is going to match the `p` character, I think you need `\p{C}` if you want to make sure that we are looking for unicode characters

**Review by peternied — CHANGES_REQUESTED** [review]

**Review by peternied — COMMENTED** [review]
Looks good!

**Review by peternied — APPROVED** [review]

File: public/apps/account/test/__snapshots__/account-nav-button.test.tsx.snap [hunk 4]
```diff
@@ -16,7 +16,7 @@ exports[`Account navigation button renders 1`] = `
     id="actionsMenu"
     isOpen={false}
     onClick={[Function]}
-    ownFocus={false}
+    ownFocus={true}
```
Comment (davidlago) [inline public/apps/account/test/__snapshots__/account-nav-button.test.tsx.snap:19]: maybe this was asked before, but why this change?

File: public/apps/configuration/panels/audit-logging/_index.scss [hunk 5]
```diff
@@ -1,11 +1,11 @@
...
 }
 
 .form-row {
-  max-width: 800px
+  max-width: 800px;
 }
 
 pre code {
-  color: #666666;
+  color: #666;
```
Comment (davidlago) [inline public/apps/configuration/panels/audit-logging/_index.scss:10]: 👿

File: server/auth/types/basic/routes.ts [hunk 6]
```diff
@@ -171,7 +171,7 @@ export class BasicAuthRoutes {
             ? this.coreSetup.http.basePath.serverBasePath
             : '/';
           const requestQuery = request.url.query as ParsedUrlQueryParams;
-          if (requestQuery.nextUrl !== undefined) {
+          if (requestQuery?.nextUrl !== undefined) {
```
Comment (peternied) [inline server/auth/types/basic/routes.ts:174]: I've created https://github.com/opensearch-project/security-dashboards-plugin/issues/936 to follow up, we should stop using the legacy url.query method

File: public/apps/account/test/__snapshots__/account-nav-button.test.tsx.snap (hunk 4 above)
Comment (DarshitChanpura) [reply public/apps/account/test/__snapshots__/account-nav-button.test.tsx.snap:19]: this is because the unit test were failing due to jest snapshot mismatch... this is something worth the discussion.

**Review by davidlago — APPROVED** [review]

File: public/apps/account/test/__snapshots__/account-nav-button.test.tsx.snap (hunk 4 above)
Comment (peternied) [reply public/apps/account/test/__snapshots__/account-nav-button.test.tsx.snap:19]: Note, this control was likely modified by a change in the styles from OpenSearch-Dashboards.  We should follow up and confirm that our UI looks clean and behaves as expected between 1.3.X and this 2.0.0 version.

I've created https://github.com/opensearch-project/security-dashboards-plugin/issues/938 for following up on this.
